File: AST/output/program expected outputs/C Progs/Draw.c
#include <stdio.h>

void printStar();
void printStar(){
	printf("  *  \n");

	printf("*****\n");

	printf(" *** \n");

	printf("*****\n");

	printf("  *  \n");

}

void printTriangle(int );
void printTriangle(int  a){
	int i, j, k;

	for (i = 0;i <= a;i++){
		for (j = a;j >= 0;j--){
			printf(" ");

		}

		for (k = 0;k < 2 * i + 1;k++){
			printf("*");

		}

		printf("\n");

	}

}

void printRectangle(int , int );
void printRectangle(int  n, int  l){
	int i, j;

	for (i = 0;i < n;i++){
		for (j = 0;j < l;j--){
			if(i == 0 || i == n - 1) {
				printf("*");
			}
			else {
				if(j == 0 || j == l - 1) {
					printf("*");
				}
				else {
					printf(" ");
				}

			}


		}

		printf("\n");

	}

}

int  main(){
	int a;

	LOOP:

	printf("Enter a number\n");

	scanf("%d", &a);

	printf("a = %d\n", a);

	switch (a) {
		case 0:
			printStar();
			break;

		case 1:
			printTriangle(5);
			break;

		case 2:
			printRectangle(4, 5);
			break;

		case 3:
			printf("END\n");
			break;

		default:
			goto LOOP;

	}

	return 0;

}

File: AST/output/program expected outputs/P Progs/Draw.py
import sys

def printStar():
	print("  *  \n")

	print("*****\n")

	print(" *** \n")

	print("*****\n")

	print("  *  \n")


def printTriangle(a):
	i, j, k = 0, 0, 0

	for i in range(a + 1):
		for j in range(a + 1):
			print(" ")

		for k in range(2 * i + 1):
			print("*")

		print("\n")


def printRectangle(n, l):
	i, j = 0, 0

	for i in range(n):
		while (j < l):
			if(i == 0 or i == n - 1):
				print("*")

			else:
				if(j == 0 or j == l - 1):
					print("*")

				else:
					print(" ")

			j -= 1

		print("\n")


def main():
	a = ""

	print("Enter a number\n")

	a = input()

	print("a = " + a + "\n")

	if(a == 0):
		printStar()

	else:
		if(a == 1):
			printTriangle(5)

		else:
			if(a == 2):
				printRectangle(4, 5)

			else:
				if(a == 3):
					print("END\n")

	return 


